mlflow/utils/file_utils.py

```diff
@@ -99,11 +100,11 @@ def mkdir(root, name=None):  # noqa
     """
     target = os.path.join(root, name) if name is not None else root
     try:
-        if not exists(target):
-            os.makedirs(target)
-            return target
+        os.makedirs(target)
     except OSError as e:
-        raise e
+        if e.errno != errno.EEXIST:
```

Comment:
This is a good change! Can we add one more check here that `target`, if it exists is a directory? If not, it should throw...
Also please could you add a unit test that throws if `target` is a file

Reply:
Updated!